torch/_dynamo/variables/torch.py

```diff
@@ -1179,6 +1207,71 @@ def speculate_branch(branch):
             )
             r = body_r.as_proxy().node.meta["example_value"]
             example_value = r
+        elif self.value.__name__ in (
+            "trampoline_autograd_fwd",
+            "trampoline_autograd_bwd",
+            "trampoline_autograd_apply",
+        ):
+            from . import AutogradFunctionVariable, UserFunctionVariable
+
+            pre_side_effects = tx.output.side_effects.clone()
+            always_restore = self.value.__name__ == "trampoline_autograd_bwd"
+            if (
+                self.value.__name__ == "trampoline_autograd_bwd"
+                or self.value.__name__ == "trampoline_autograd_fwd"
+            ):
+                fn = UserFunctionVariable(self.value, source=self.value._source)
+            else:
+                fn = TorchVariable(self.value)
+            checkpoint = tx.copy_graphstate()
+            pre_guards = tx.output.guards
+            graph_checkpoint = tx.output.graph
+            (
+                body_r,
+                body_graph,
+                body_lifted_freevars,
+            ) = speculate_subgraph(
+                tx,
+                fn,
+                [
+                    *args,
+                ],
+                graph_checkpoint,
+                checkpoint,
+                # Backwards should never, ever be stored!
+                always_restore=always_restore,
+            )
+            post_guards = tx.output.guards
+            if body_lifted_freevars:
+                for freevar in body_lifted_freevars:
+                    if "whitelisted" not in freevar.node.meta:
```

Comment:
The "whitelisted" thing feels super hacky but I don't have any better suggestions lol

Reply:
Yea its a little unfortunate... but not egregious I guess?

Reply:
also, nit: call it 'allowlisted' or 'came_from_saved_tensors'

Reply:
agreed.